homeassistant/components/ecobee/climate.py

```diff
@@ -727,6 +749,58 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, sensors: list[str], preset_mode: str | None = None
+    ) -> None:
+        """Set the sensors used on a climate for a thermostat."""
+        if preset_mode is None:
+            preset_mode = self.preset_mode
+
+        # Check if climate is an available preset option.
+        elif preset_mode not in self._preset_modes.values():
+            msg = f"Invalid climate name, available options are: {', '.join(self.preset_modes)}"
+            raise ServiceValidationError(msg)
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        for sensor in sensors:
+            sensor_registry = device_registry.async_get(sensor)
+            if sensor_registry and sensor_registry.name:
+                sensor_names.append(sensor_registry.name)
+
+        # Ensure sensors provided are available for thermostat.
+        if not set(sensor_names).issubset(set(self._sensors)):
+            msg = f"Invalid sensor for thermosat, available options are: {', '.join(self._sensors)}"
+            raise ServiceValidationError(msg)
+
+        # Check if sensors are currently used on the climate for the thermostat.
+        current_sensors_in_climate = self._sensors_in_preset_mode(preset_mode)
+        if set(sensor_names) == set(current_sensors_in_climate):
+            msg = f"This action would not be an update, current sensors on climate ({preset_mode}) are: {', '.join(current_sensors_in_climate)}"
+            raise ServiceValidationError(msg)
```

Comment:
Good callout on raising a log message instead of an error. An error does not need to be called since the current state is the desired state. I have added a debug message.

I like the idea of keeping track of the sensors. It prevents unnecessary api calls and in the future could be an attribute on the thermostat.